### ISSUE TITLE
train the model only on the real data can achieve good result

### ISSUE THREAD

**Issue by PengfeiRen96** [poier/murauer#1]
Hi Georg,

I tried to re-implement you excellent work on my machine. 
First,I found that train the ResNet50 only on the real data can achieve good reuslt(12.02mm) without the ViewDecoder and the discriminator. It looks like different from your paper results(14.7mm). 
Second, i tried some different setup to train the ResNet50 only on real data.  I achieved best results (10.63mm) under below  setup: sgd, batchsize=128, lr=0.3, decay_facor=0.1,step_size=20epochs, max_epoch=80.
I'm afraid some of the details of my experiment are wrong. Can you try my setup on your machine?

Cheers.

**Comment by poier** [comment]
Hi,

thanks for your comment.
Yes, we also get better results with the published code than the ones reported in the paper. As mentioned in the readme, the two main changes yielding improved results were some bug fixes (esp. with the non-linearities) and using the new PyTorch version (v1.0.1). (The results of the published models given in the readme also show this). I also believe that more tuning towards the results on the test set have the potential to improve the corresponding results (we, obviously, didn't do this for the paper :-)). 
Thank you also for sharing a better set of hyperparameters. I'll try it as soon as I can. 

Best regards,
Georg

**Comment by poier** [comment]
Hi,
I ran several experiments with your suggested hyper-parameter settings with our code now. The error on the test set were always between 15 and 16mm (with PyTorch v0.4.1 and v1.0.1). With the settings, which worked best for the paper (lr=0.00033, bs=256, exponential lr-decay as described in the paper), we get an error or ~13.5mm with the updated code and current PyTorch version v1.0.1.
That is, I could not reproduce your results. Can you provide some more details about your re-implementation or publish your code?

Best,
Georg

**Comment by PengfeiRen96** [comment]
Hi Georg,
Thank you for your attention to this matter. I will clean my code and publish it as soon as I can. I tested this  hyper-parameter on ICVL and MSRA. we also got good result. This made me wonder that some of the previous methods did not fully utilize the capabilities of the network.

best，
pengfei Ren

**Comment by poier** [comment]
Hi Pengfei,
I see. Looking forward to it.

Best,
Georg

**Comment by PengfeiRen96** [comment]
Hi Georg,
I am very sorry for such a late reply.I am worried about finding a job for my girlfriend recently.I have update my code which is a simple version. https://github.com/RenFeiTemp/murauer/tree/master/source/my_net

Best,
Georg

**Comment by poier** [comment]
Hi Pengfei,

sorry that it took me so long to get back to it. Finally, I got to check your code and did some experiments. The single main difference I spoted between the implementations (besides the mentioned hyperparameters, jittering of test samples and a minor issue at evaluation) is that you are using a Huber Loss. 
I did some experiments to see the impact of this. As mentioned earlier the mean error when using only real data is ~13.5mm with our implementation. When using the Huber Loss (with your parameters for the Huber) it goes down to ~11mm, however when I use your hyperparameter settings (sgd, lr-decay, bs, ...) it is ~12.5mm.
On the other hand, with your implementation the mean error is ~10.6mm and when I use the mean squared error as a loss instead of the Huber it goes up to 11.3mm.
So, there's still some (seemingly) significant difference. It would be great if you could tell me what else you changed.
One side note: When checking your code I found an issue with the evaluation - with very minor impact, however. The code averages the error within each batch, aggregates over batches and averages again. This assumes that the batch size is always the same, which is usually not the case. The last batch of the test set is usually smaller (depending on the batch size, of course) and, hence, its samples get a higher weight. Nevertheless, - as stated above - the impact of this is negligible.

Best,
Georg

**Comment by poier** [comment]
Hi Pengfei,

working on other topics, it came to my mind that I probably never checked whether we used the same weight decay settings. :-) And indeed, you set the weight decay factor to 0.0001, while it was 0.001 in my setup. Obviously, this was the difference I missed and after changing it to your setting, I get the same results as you do when using your hyper-parameter settings.

Best,
Georg

**Comment by PengfeiRen96** [comment]
Hi Georg,

Very sorry, I mistakenly thought that I have already replied to you. Actually, I have done some experiments with your parameter configuration. I also didn't reproduce your results. Thank you very much for your patience. What's more, I found that resnet18 can achieve almost the same performance as point cloud networks and 3D networks only by carefully setting the hyperparameters. So I think it might be redundant to use point clouds or 3D networks for 2.5D depth data. I will update my code as soon as possible. 
I apologize again for the late reply.

Best,
Pengfei
